### PR TITLE
Update frontend test and Ivy test docs

### PR DESCRIPTION
**Pull request by CatB1t**

### CODE REVIEW

File: docs/partial_source/deep_dive/16_ivy_frontends.rst [hunk 1]
```diff
@@ -420,6 +417,11 @@ not necessary to uniquely flag every single NumPy function as supporting only CP
 as this is a limitation of the entire framework, and this limitation is already
 `globally flagged <https://github.com/unifyai/ivy/blob/6eb2cadf04f06aace9118804100b0928dc71320c/ivy/functional/backends/numpy/__init__.py#L21>`_.
 
+Another common case, If the function supports a data type, but a backend framework does not. the frontend function `jax.lax.cumprod <https://github.com/unifyai/ivy/blob/6e80b20d27d26b67a3876735c3e4cd9a1d38a0e9/ivy/functional/frontends/jax/lax/operators.py#L111>`_
```
Comment (djl11) [inline docs/partial_source/deep_dive/16_ivy_frontends.rst:420]: Double check the grammar here. You have a capital letter after a comma, and then a non-capital letter after a comma. In general it is quite hard to follow along with what the overall sentence is saying. Please try and restructure it for maximal clarity :)

**Review by djl11 — CHANGES_REQUESTED** [review]

File: docs/partial_source/deep_dive/16_ivy_frontends.rst [hunk 2]
```diff
@@ -520,18 +522,17 @@ instance, and then passes the key and value as inputs to the corresponding funct
...
+
     @handle_cmd_line_args
     @given(
-        dtype_x_axis=_dtype_x_axis(available_dtypes=helpers.get_dtypes("float")),
+        dtype_x_axis=helpers.dtype_values_axis(available_dtypes=helpers.get_dtypes("float")),
         dtype=helpers.get_dtypes("float", full=False, none=True),
-        keep_dims= st.one_of (st.booleans(), Novalue),
-        initial=st.one_of(st.floats(), Novalue),
+        keep_dims=st.booleans(),
+        initial=st.one_of(st.floats(), st.none()),
```
Comment (djl11) [inline docs/partial_source/deep_dive/16_ivy_frontends.rst:535]: why have you removed the `Novalue` stuff? This was one of the points mentioned in the sentence above?

**Review by djl11 — CHANGES_REQUESTED** [review]

File: docs/partial_source/deep_dive/16_ivy_frontends.rst (hunk 2 above)
Comment (CatB1t) [reply docs/partial_source/deep_dive/16_ivy_frontends.rst:535]: I just used the recent version of it here (I wasn't sure about what happened to it) : https://github.com/unifyai/ivy/blob/8554cd304f9a35d39693e03878aab62a97acbcef/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py#L18

Comment (CatB1t) [reply docs/partial_source/deep_dive/16_ivy_frontends.rst]: I've temporary reverted the changes for it.

**Review by djl11 — APPROVED** [review]

**Comment by djl11** [comment]
just check with @Daniel4078 why the NoValue stuff was removed, and if it is meant to be removed then maybe we can remove this framework-specific class explanation from the docs entirely.
